### PR TITLE
Upgrade grafter, simplify deps, switch to tools.build and circleci

### PR DESCRIPTION
**Pull request by RickMoynihan**
This PR does a number of things:

1. It simplifies the testing in this project so it does not require a running triple store service, by using an in memory SPARQL repository.
2. It runs the tests on CI resolving issue #9 
3. It switches the build to use `tools.deps` and `tools.build` rather than `leiningen`.
4. It switches our CI from travis to circleci.
5. It upgrades grafter (and RDF4j from an old version to a recent 4.x release) from 2.x to a candidate `3.0.0-SNAPSHOT` version
6. It publishes this artifact to clojars
7. It removes the usage/complexity of duct config in the test setup

NOTE: I haven't yet finalised a grafter `3.0.0` release, as I'd like to upgrade it, and test it in a number of apps first which use this dependency.  This PR goes with this [grafter PR](https://github.com/Swirrl/grafter/pull/187)

*NOTE*: The maven coordinates for this project had to change, as it used an "unverified" group-id which was deprecated a year or two ago on clojars.  This means consumers need to be careful not to have any older versions of `grafter.db` or `grafter` in their transitive dependency trees, as `tools.deps` won't be able to resolve the versions correctly, and users may end up with multiple versions colliding on their classpath in an unpredictable fashion.

### CODE REVIEW

**Comment by RickMoynihan** [comment]
I should add I did originally try to separate out the grafter version changes, from the rest of the changes; but it turned out:

1. CI previously didn’t these tests because they depended on a running sparql service, and nobody had the stomach for previously adding that as a dependency to CI.  So I wanted to switch to an in memory sparql repository.
2. The old RDF4j in memory sparql repository used by the earlier release of grafter returned different results for a number of tests, however these issues appear to have been fixed in more recent versions.

**Comment by scottlowe** [comment]
@RickMoynihan I got caught out by the Java 11 requirement on my local machine. This is a deliberate dependancy choice so that we can move to the latest `rdf4j` right? 

Is suppose that it must be, considering that you chose an `openjdk-11` CircleCI image and we plan to move to Java 11 a.s.a.p.

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,62 @@
+version: 2.1
+
+# Define a job to be invoked later in a workflow.
+# See: https://circleci.com/docs/2.0/configuration-reference/#jobs
+jobs:
+  test:
+    docker:
+      - image: cimg/clojure:1.10
```
Comment (scottlowe) [inline .circleci/config.yml:8]: Why not `cimg/clojure:1.10.3-openjdk-11.0` to match a more specific Clojure version and target JDK ?

That would then match the exact versions used in the Docker image specified in the `deploy` section.

**Comment by RickMoynihan** [comment]
> @RickMoynihan I got caught out by the Java 11 requirement on my local machine. This is a deliberate dependancy choice so that we can move to the latest rdf4j right?

Ahh thanks for spotting this, I'd actually forgotten that was the case.  Basically I copied and edited the circleci/tools.deps/build setup from the one I wrote for matcha.  

So this was something I half forgot, and is something that definitely needs to be made more prominent in terms of implications, but yes the latest RDF4j's and grafter require JDK11+.  This is of course an additional complication, particularly when it comes to deployments.

TBH if we're going to update the JDK deps as part of this, we might as well try and push it to JDK 17 which is the latest LTS release.

File: .circleci/config.yml (hunk 1 above)
Comment (RickMoynihan) [reply .circleci/config.yml:8]: Thanks for spotting will look at this...

**Pull request merged**